test/foundry/FleekERC721/FleekERC721.access.t.sol

```diff
@@ -0,0 +1,444 @@
+        address randomAddress = address(99);
+
+        // CollectionOwner
+        vm.startPrank(collectionOwner);
+        expectRevertWithTokenRole();
+        CuT.grantTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        expectRevertWithTokenRole();
+        CuT.revokeTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        vm.stopPrank();
+
+        // CollectionController
+        vm.startPrank(collectionController);
+        expectRevertWithTokenRole();
+        CuT.grantTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        expectRevertWithTokenRole();
+        CuT.revokeTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        vm.stopPrank();
+
+        // TokenOwner
+        vm.startPrank(tokenOwner);
+        CuT.grantTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        assertTrue(CuT.hasTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress));
+        CuT.revokeTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        assertFalse(CuT.hasTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress));
+        vm.stopPrank();
+
+        // TokenController
+        vm.startPrank(tokenController);
+        expectRevertWithTokenRole();
+        CuT.grantTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        expectRevertWithTokenRole();
+        CuT.revokeTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        vm.stopPrank();
+
+        // AnyAddress
+        vm.startPrank(anyAddress);
+        expectRevertWithTokenRole();
+        CuT.grantTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        expectRevertWithTokenRole();
+        CuT.revokeTokenRole(tokenId, FleekAccessControl.Roles.Controller, randomAddress);
+        vm.stopPrank();
+    }
+
+    function test_mint() public {
+        address randomAddress = address(99);
+
+        // CollectionOwner
+        vm.startPrank(collectionOwner);
+        mintDefault(randomAddress);
+        vm.stopPrank();
+
+        // CollectionController
+        vm.startPrank(collectionController);
+        expectRevertWithCollectionRole();
+        mintDefault(randomAddress);
+        vm.stopPrank();
+
+        // TokenOwner
+        vm.startPrank(tokenOwner);
+        expectRevertWithCollectionRole();
+        mintDefault(randomAddress);
+        vm.stopPrank();
+
+        // TokenController
+        vm.startPrank(tokenController);
+        expectRevertWithCollectionRole();
+        mintDefault(randomAddress);
+        vm.stopPrank();
+
+        // AnyAddress
+        vm.startPrank(anyAddress);
+        expectRevertWithCollectionRole();
+        mintDefault(randomAddress);
+        vm.stopPrank();
+    }
+
+    function test_tokenURI() public {
+        // ColletionOwner
+        vm.prank(collectionOwner);
+        CuT.tokenURI(tokenId);
```

Comment:
I don't think it's necessary to check all of these addresses if  `anyAddress` works.